packages/logger-plugin/src/log-writer.ts

```diff
@@ -0,0 +1,62 @@
+import { NgxsLoggerPluginOptions } from './symbols';
+export class LogWriter {
+  private logger: any;
+
+  constructor(private options: NgxsLoggerPluginOptions) {
+    this.options = this.options || <any>{};
+    this.logger = options.logger || console;
+  }
+
+  startGroup(message: string) {
+    const startGroupFn = this.options.collapsed
+      ? this.logger.groupCollapsed
+      : this.logger.group;
+    try {
+      startGroupFn.call(this.logger, message);
+    } catch (e) {
+      console.log(message);
+    }
+  }
+
+  endGroup() {
+    try {
+      this.logger.groupEnd();
+    } catch (e) {
+      this.logger.log('—— log end ——');
+    }
+  }
+
+  logGrey(title: string, payload: any) {
+    this.log(title, 'color: #9E9E9E; font-weight: bold', payload);
+  }
+
+  logGreen(title: string, payload: any) {
+    this.log(title, 'color: #4CAF50; font-weight: bold', payload);
+  }
+
+  logRedish(title: string, payload: any) {
+    this.log(title, 'color: #FD8182; font-weight: bold', payload);
+  }
+
+  log(title: string, color: string, payload: any) {
+    if (this.isIE()) {
+      this.logger.log(title, payload);
+    } else {
+      this.logger.log('%c ' + title, color, payload);
+    }
+  }
+
+  isIE(): boolean {
```

Comment:
There is no support anywhere else

*Internet Explorer (IE) 10 will reach end of support on January 31, 2020. From this point forward, IE 11 will be the only supported version of Internet Explorer on Windows Server 2012 and Windows Embedded 8 Standard. IE 11 is now available for piloting via Microsoft Update Catalog

Reply:
Interesting. We will need to keep this in mind.
This is a refactoring, so I will not remove code that is possibly defining a behavior.

Reply:
ok